apartmenttherapy/oxygen-lib-ui
# Henry/new pin

#### What does this PR do & why?
Refactors the custom `PinItButton` component to use the new Save button documented here:
https://developers.pinterest.com/docs/widgets/save/?

To that end 
- takes out the custom click handler as the new Pinterest script (more about this on the Oxygen side) takes care of what happens on click
- swaps the <button> for an <a> per the docs
- moves `PinItButton` out of `LazyLoad` so that the Pinterest script can process the button on page load.

#### How can this be manually tested?
You will need the companion `oxygen` PR: https://github.com/apartmenttherapy/oxygen/pull/1025

Pull and go to any post page. You should be able to click the Pin button on standalone image or gallery image, and you'll see a pop-up that lets you add that image to a Pinterest board.

#### Any questions or concerns about this change?
Unrelated note: clicking on the Pinterest button on a gallery slide expands the gallery. I will take care of that in a separate PR

#### Deployment Notes

_(e.g. Does this require a migration or special infrastructure change? Clearing the cache? Third party service?)_

#### Link to story card and other documentation
https://app.asana.com/0/937985426519793/919759511279010/f

#### GIF
![pin_placement](https://user-images.githubusercontent.com/5115651/51133339-bf4ce680-1802-11e9-97d0-69681427ef55.jpg)


#### Have you remembered to...?
